### PR TITLE
OPEN-72 property value arithmetics

### PR DESCRIPTION
**Pull request by anton-larsen-persequor**

### CODE REVIEW

File: ran-core/src/main/java/io/ran/Property.java [hunk 1]
```diff
@@ -344,4 +383,235 @@ public Property<T> getProperty() {
...
+		public abstract NumberValue<V> divide(NumberValue<?> numberValue);
+
+		private static class ByteNumberValue extends NumberValue<Byte> {
+			public ByteNumberValue(Object value) {
+				super(((Number)value).byteValue());
+			}
+
+			@Override
+			public NumberValue<Byte> add(NumberValue<?> other) {
+				return new ByteNumberValue(value() + PropertyValue.getNumberValue(Byte.class,other.value()).value());
```
Comment (hampus-ramstrom-psqr) [inline ran-core/src/main/java/io/ran/Property.java:442]:
```suggestion
				return new ByteNumberValue(value() + PropertyValue.getNumberValue(Byte.class, other.value()).value());
```
Nitpick, apply everywhere.

File: ran-core/src/main/java/io/ran/Property.java [hunk 2]
```diff
@@ -344,4 +383,235 @@ public Property<T> getProperty() {
...
+				return new BigDecimalNumberValue(value().multiply(PropertyValue.getNumberValue(BigDecimal.class,other.value()).value()));
+			}
+
+			@Override
+			public NumberValue<BigDecimal> divide(NumberValue<?> other) {
+				return new BigDecimalNumberValue(value().divide(PropertyValue.getNumberValue(BigDecimal.class,other.value()).value()));
+			}
+		}
+	}
+
```
Comment (hampus-ramstrom-psqr) [inline ran-core/src/main/java/io/ran/Property.java:616]:
```suggestion
```
Nitpick.

File: ran-core/src/test/java/io/ran/PropertyValueTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package io.ran;
+
+import io.ran.token.Token;
+import org.junit.Test;
+
+import java.lang.reflect.Constructor;
```
Comment (hampus-ramstrom-psqr) [inline ran-core/src/test/java/io/ran/PropertyValueTest.java:6]:
```suggestion
```
Unused.

File: ran-core/src/test/java/io/ran/PropertyValueTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import java.lang.reflect.InvocationTargetException;
+
+import static org.junit.Assert.*;
+
+public class PropertyValueTest {
+	private Property property = Property.get(Token.of("my"));
+
+	@Test
+	public void getNumericValue_simple_int() {
```
Comment (hampus-ramstrom-psqr) [inline ran-core/src/test/java/io/ran/PropertyValueTest.java:15]: Not sure if the name reflects the test here, another suggestion could be `propertyValue_simple_int_operations`

**Review by hampus-ramstrom-psqr — APPROVED** [review]
LGTM, some minor comments.

**Comment by hampus-ramstrom-psqr** [comment]
Maybe it would make sense for me to test this out in relation to [SCRM-350](https://persequor.atlassian.net/browse/SCRM-350) before we merge it into `main`.

File: ran-core/src/main/java/io/ran/Property.java [hunk 5]
```diff
@@ -325,6 +360,10 @@ public int hashCode() {
 		}
 	}
 
+	private Clazz getClazz() {
+		return type;
+	}
```
[the remark this answers is not in the document]
Comment (anton-larsen-persequor) [reply ran-core/src/main/java/io/ran/Property.java:365]: But it is useful

[the remark this answers is not in the document]
Comment (hampus-ramstrom-psqr) [reply ran-core/src/main/java/io/ran/Property.java]: 4d8dbb7400a1150062e9a62104b9a716f85e8436

File: ran-core/src/main/java/io/ran/Property.java (hunk 2 above)
Comment (hampus-ramstrom-psqr) [reply ran-core/src/main/java/io/ran/Property.java]: 4d8dbb7400a1150062e9a62104b9a716f85e8436

File: ran-core/src/test/java/io/ran/PropertyValueTest.java (hunk 4 above)
Comment (hampus-ramstrom-psqr) [reply ran-core/src/test/java/io/ran/PropertyValueTest.java]: 4d8dbb7400a1150062e9a62104b9a716f85e8436

**Comment by hampus-ramstrom-psqr** [comment]
> Maybe it would make sense for me to test this out in relation to [SCRM-350](https://persequor.atlassian.net/browse/SCRM-350) before we merge it into `main`.

This is not doable (at least not in a simple manner). So will be merging it down and then propagate the changes into Saga.

**Pull request merged**